src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -174,73 +133,111 @@ export default Utils.withCancellationSignal(class ShareWorkspaceModal extends Co
-          autoFocus: !!numAdditions,
+          autoFocus: email === lastAddedEmail,
           value: aclItem,
-          onChange: v => this.setState(_.set(['acl', index], v)),
+          onChange: v => setAcl(_.set([index], v)),
           disabled,
           maxAccessLevel: workspace.accessLevel
         })
       ]),
       !disabled && h(Link, {
-        onClick: () => this.setState({ acl: _.remove({ email }, acl) })
+        onClick: () => setAcl(_.remove({ email }))
       }, [icon('times', { size: 20, style: { marginRight: '0.5rem' } })])
     ])
   }
 
-  async componentDidMount() {
-    const { workspace: { workspace: { namespace, name } }, onDismiss, signal } = this.props
 
-    try {
-      const [{ acl }, shareSuggestions, groups] = await Promise.all([
-        Ajax(signal).Workspaces.workspace(namespace, name).getAcl(),
-        Ajax(signal).Workspaces.getShareLog(),
-        Ajax(signal).Groups.list()
-      ])
-
-      const fixedAcl = _.flow(
-        _.toPairs,
-        _.map(([email, data]) => ({ email, ...data })),
-        _.sortBy(x => -Utils.workspaceAccessLevels.indexOf(x.accessLevel))
-      )(acl)
-
-      this.setState({
-        acl: fixedAcl,
-        originalAcl: fixedAcl,
-        groups,
-        shareSuggestions,
-        loaded: true
-      })
-    } catch (error) {
-      onDismiss()
-      reportError('Error looking up collaborators', error)
+  // Lifecycle
+  Utils.useOnMount(() => {
+    const load = async () => {
+      try {
+        const [{ acl }, shareSuggestions, groups] = await Promise.all([
+          Ajax(signal).Workspaces.workspace(namespace, name).getAcl(),
+          Ajax(signal).Workspaces.getShareLog(),
+          Ajax(signal).Groups.list()
+        ])
+
+        const fixedAcl = _.flow(
+          _.toPairs,
+          _.map(([email, data]) => ({ email, ...data })),
+          _.sortBy(x => -Utils.workspaceAccessLevels.indexOf(x.accessLevel))
+        )(acl)
+
+        setAcl(fixedAcl)
+        setOriginalAcl(fixedAcl)
+        setGroups(groups)
+        setShareSuggestions(shareSuggestions)
+        setLoaded(true)
+      } catch (error) {
+        onDismiss()
+        reportError('Error looking up collaborators', error)
+      }
     }
-  }
 
-  async save() {
-    const { workspace: { workspace: { namespace, name } }, onDismiss } = this.props
-    const { acl, originalAcl } = this.state
-
-    const aclEmails = _.map('email', acl)
-    const needsDelete = _.remove(entry => aclEmails.includes(entry.email), originalAcl)
-    const numAdditions = _.filter(({ email }) => !_.some({ email }, originalAcl), acl).length
-    const eventData = { numAdditions, ...extractWorkspaceDetails(this.props.workspace) }
-
-    const aclUpdates = [
-      ..._.flow(
-        _.remove({ accessLevel: 'PROJECT_OWNER' }),
-        _.map(_.pick(['email', 'accessLevel', 'canShare', 'canCompute']))
-      )(acl),
-      ..._.map(({ email }) => ({ email, accessLevel: 'NO ACCESS' }), needsDelete)
-    ]
+    load()
```

Comment:
Do we need an await here since load() is async? Also how come the old version didn't need the async?

Reply:
So, we can't await, since this is inside a `useEffect` (that's what `useOnMount` actually is; we can only kick it off, it has to await internally. The old version was async, look at the old line 190.